TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/service/implementation/AuthenticationServiceImplementation.java

```diff
@@ -109,4 +110,27 @@ public boolean logOut(Long userId) {
         // Returns false due to an exception thrown by findById
         return false;
     }
+
+    @Override
+    public AuthenticationResponse resetPassword(UserResetPasswordDTO userDTO) {
+        try {
+            String email = userDTO.getEmail();
+            String oldPassword = userDTO.getOldPassword();
+            String newPassword = userDTO.getNewPassword();
+            authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(email, oldPassword));
+            var user = userRepository.findByEmail(email)
+                    .orElseThrow();
+            user.setPassword(passwordEncoder.encode(newPassword));
+            userRepository.save(user);
+
+            var jwtToken = jwtService.generateToken(user);
+            return AuthenticationResponse.builder()
+                    .token(jwtToken)
+                    .build();
+        } catch (Exception exception) {
+            log.error("{Error}", exception);
+        }
+
+        return null;
```

Comment:
W: A bit late now, but instead of returning null, we should have used exceptions: if an exception is thrown in the try-catch, you should throw it further and catch it where you call the method